### PR TITLE
Simplify sstable_directory configuration

### PR DESCRIPTION
**Pull request by xemul**
When started the sstable_directory is constructed with a bunch of booleans that control the way its process_sstable_dir method works. It's shorter and simpler to pass these booleans into method directly, all the more so there's another flag that's already passed like this.

### CODE REVIEW

**Review by avikivity — APPROVED** [review]
Don't see what RAII has to do with it.

@raphaelsc please review

**Comment by xemul** [comment]
@raphaelsc , ping

File: replica/distributed_loader.cc [hunk 1]
```diff
@@ -326,6 +322,8 @@ distributed_loader::process_upload_dir(distributed<replica::database>& db, distr
 
         lock_table(directory, db, ks, cf).get();
         sstables::sstable_directory::process_flags flags {
+            .need_mutate_level = true,
+            .enable_dangerous_direct_import_of_cassandra_counters = db.local().get_config().enable_dangerous_direct_import_of_cassandra_counters(),
         };
```
[the remark this answers is not in the document]
Comment (xemul) [reply replica/distributed_loader.cc:327]: Sure, no problem

**Comment by xemul** [comment]
Update:
- initialize process_flags.allow_loading_materialized_view always

**Review by raphaelsc — APPROVED** [review]

**Pull request closed without merging**